homeassistant/components/nederlandse_spoorwegen/sensor.py

```diff
@@ -135,30 +135,29 @@ def device_state_attributes(self):
             ),
             "departure_time_actual": self._trips[0].departure_time_actual.strftime(
                 "%H:%M"
-            ),
+            )
+            if self._trips[0].departure_time_actual is not None
+            else None,
             "departure_delay": self._trips[0].departure_time_planned
             != self._trips[0].departure_time_actual,
-            "departure_platform": self._trips[0].trip_parts[0].stops[0].platform,
-            "departure_platform_changed": self._trips[0]
-            .trip_parts[0]
-            .stops[0]
-            .platform_changed,
+            "departure_platform": self._trips[0].departure_platform_planned,
+            "departure_platform_changed": self._trips[0].departure_platform_actual,
             "arrival_time_planned": self._trips[0].arrival_time_planned.strftime(
                 "%H:%M"
             ),
-            "arrival_time_actual": self._trips[0].arrival_time_actual.strftime("%H:%M"),
+            "arrival_time_actual": self._trips[0].arrival_time_actual.strftime("%H:%M")
+            if self._trips[0].arrival_time_actual is not None
+            else None,
             "arrival_delay": self._trips[0].arrival_time_planned
             != self._trips[0].arrival_time_actual,
-            "arrival_platform": self._trips[0].trip_parts[0].stops[-1].platform,
-            "arrival_platform_changed": self._trips[0]
-            .trip_parts[0]
-            .stops[-1]
-            .platform_changed,
-            "next": self._trips[1].departure_time_actual.strftime("%H:%M"),
+            "arrival_platform": self._trips[0].arrival_platform_planned,
+            "arrival_platform_changed": self._trips[0].arrival_platform_actual,
+            "next": self._trips[1].departure_time_actual.strftime("%H:%M")
+            if self._trips[1].departure_time_actual is not None
+            else self._trips[1].departure_time_planned.strftime("%H:%M"),
```

Comment:
These conditionals are hard to read when formatted like that. I suggest you create the data with something like this:
```python
attributes = {
    "going": self._trips[0].going,
    ... other constant values ...
}

if self._trips[0].departure_time_actual is not None:
    attributes["departure_time_actual"] = self._trips[0].departure_time_actual.strftime("%H:%M")

if self._trips[1].departure_time_actual is not None:
    attributes["next"] = self._trips[1].departure_time_actual.strftime("%H:%M")
else:
    attributes["next"] = self._trips[1].departure_time_planned.strftime("%H:%M")

return attributes
```